test/test_ion_text.cpp

```diff
@@ -720,3 +720,426 @@ TEST(IonTextInt, BinaryLiterals) {
+
+    // {{ V2hhdCBhcmUgeW91IGRvaW5nLCBEYXZlPw== }}
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_BLOB, type);
+    // assert position
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 52);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 52);
+
+    // {{ '''I can't let you do that, Dave.''' }}
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_CLOB, type);
+    // assert position
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 95);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 95);
+
+    // 42
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    // assert position
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 138);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 138);
+
+}
+
+/** tests ion_reader_get_value_position for annotated values. */
+TEST(IonTextPosition, PositionOfAnnotatedValues) {
+    const char *ion_text =
+        // Column number cheat sheet:
+        // 1         2
+        // 01234567890123456
+        "  a::1 b::2 c::3 ";
+
+    hREADER  reader;
+    ION_TYPE type;
+
+    ION_ASSERT_OK(ion_test_new_text_reader(ion_text, &reader));
+
+    int64_t offset = 0;
+    int32_t line = 0;
+    int32_t col_offset = 0;
+
+    // a::1
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 2);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 2);
+
+    // b::2
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 7);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 7);
+
+    // c::3
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_INT, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 12);
+    ASSERT_EQ(line, 1);
+    ASSERT_EQ(col_offset, 12);
+
+}
+
+/** tests ion_reader_get_value_position for list elements appearing on separate lines. */
+TEST(IonTextPosition, PositionOfListElements) {
+    // this is `[1,2,3] 42` split across multiple lines with values prefixed with various tabs and spaces.
+    const char *ion_text = "\n[\n\t1,\n\t 2,\n\t  3\n]\n 42";
+
+    hREADER  reader;
+    ION_TYPE type;
+
+    ION_ASSERT_OK(ion_test_new_text_reader(ion_text, &reader));
+
+    int64_t offset = 0;
+    int32_t line = 0;
+    int32_t col_offset = 0;
+
+    ION_ASSERT_OK(ion_reader_next(reader, &type));
+    ASSERT_EQ(tid_LIST, type);
+    ION_ASSERT_OK(ion_reader_get_value_position(reader, &offset, &line, &col_offset));
+    ASSERT_EQ(offset, 1);
+    ASSERT_EQ(line, 2);
```

Comment:
So lines start at 1, while offsets start at 0. I think that's fine, but it would be good to make sure it's documented on the function.

Reply:
added.